libnopegl/src/math_utils.c

```diff
@@ -118,6 +118,12 @@ void ngli_vec4_lerp(float *dst, const float *v1, const float *v2, float c)
     dst[3] = NGLI_MIX_F32(a[3], b[3], c);
 }
 
+void ngli_vec4_perspective_div(float *dst, float *v)
+{
+    float r[] = {v[0] / v[3], v[1] / v[3], v[2] / v[3], v[3] / v[3]};
+    memcpy(dst, r, sizeof(r));
+}
```

Comment:
1. Isn't this `NGLI_VEC3_SCALE(v, v[3])`?
2. Missing a `const` for `v` and `r`
3. Isn't this very specific?

Reply:
1. yes, it's `NGLI_VEC4_SCALE(v, 1.f/v[3])` yes
2. added missing const
3. it's a function that is quite useful when you want to apply the perspective on the CPU side, it could be useful for bounding boxes computation or debugging